lib/resources/groups.rb

```diff
@@ -154,6 +158,35 @@ def groups
     end
   end
 
+  # OSX uses opendirectory for groups, so `/etc/group` may not be accurate
+  class DarwinGroup < GroupInfo
+    def groups
+      raw_output = inspec.command('dscacheutil -q group').stdout
+      raw_group_info = raw_output.split("\n\n")
+
+      group_info = []
+      regex = /^([^:]*?)\s*:\s(.*?)\s*$/
+      raw_group_info.each do |data|
+        group_info << inspec.parse_config(data, assignment_regex: regex).params
+      end
+
+      match_etc_group_output(group_info)
+    end
+
+    # Converts the `dscacheutil` groups to match `inspec.etc_group.entries`
+    def match_etc_group_output(dscacheutil_groups)
```

Comment:
What's the rationale for breaking this out into a separate method?

Reply:
I wanted to isolate the logic to keep the `groups` method clean. In hindsight, I should have made it a private method. I'll move the logic back into `groups` though since it's only a few lines an a comment will suffice.